assets/js/base/context/shared/event-emit/emitters.ts

```diff
@@ -50,21 +49,23 @@ export const emitEvent = async ( observers, eventType, data ) => {
  *
  * @param {Object} observers The registered observers to omit to.
  * @param {string} eventType The event type being emitted.
- * @param {*}      data      Data passed along to the observer when it is
- *                           invoked.
+ * @param {*}      data      Data passed along to the observer when it is invoked.
  *
- * @return {Promise} Returns a promise that resolves to either boolean or the
- *                   return value of the aborted observer.
+ * @return {Promise} Returns a promise that resolves to either boolean or the return value of the aborted observer.
  */
-export const emitEventWithAbort = async ( observers, eventType, data ) => {
+export const emitEventWithAbort = async (
+	observers: EventObserversType,
+	eventType: string,
+	data: unknown
+): Promise< unknown > => {
 	const observersByType = getObserversByPriority( observers, eventType );
 	for ( const observer of observersByType ) {
 		try {
 			const response = await Promise.resolve( observer.callback( data ) );
-			if ( typeof response !== 'object' ) {
+			if ( typeof response !== 'object' || response === null ) {
 				continue;
 			}
-			if ( typeof response.type === 'undefined' ) {
+			if ( ! response.hasOwnProperty( 'type' ) ) {
```

Comment:
`if (!response?.type) {}` could also work here

Reply:
Typescript complains that type does not exist on response with your suggestion. `hasOwnProperty` was the only way I could suppress that warning.

Reply:
I had a hunch that Ts might be behind this one